types/three/three-core.d.ts

```diff
@@ -5255,6 +5271,14 @@ export class Mesh extends Object3D {
     clone(): Mesh;
 }
 
+/**
+ * An intermediate material type that casts more precisely the possible materials assignable to a [[Points]] object.
+ *
+ * // Todo:
+ * // - can [[Points]] take in an array of materials ?
+ */
+export type PointsMaterialType = PointsMaterial | ShaderMaterial | MeshDepthMaterial;
```

Comment:
Why is `MeshDepthMaterial` a part of this union?

Reply:
As for lines, MeshDepthMaterial is used as an overrideMaterial in the scene for postprocessing. However as opposed to the lines I don't remember what the result was. MeshDepthMaterial should not be able to render points correctly, it should be missing `gl_PointSize` in the shader, as for the other Line/Mesh materials.

This overrideMaterial situation does raise some questions that may be answered on the ThreeJS issue you opened.